api/v1beta1/temporalappworker_types.go

```diff
@@ -0,0 +1,101 @@
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package v1beta1
+
+import (
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// TemporalAppWorkerSpec defines the desired state of TemporalAppWorker
+type TemporalAppWorkerSpec struct {
+	// Version defines the app worker version.
+	Version string `json:"version"`
+	// Image defines the temporal ui docker image the instance should run.
+	Image string `json:"image"`
+	// Number of desired replicas. Default to 1.
+	// +kubebuilder:validation:Minimum=1
+	// +optional
+	Replicas *int32 `json:"replicas"`
+	// An optional list of references to secrets in the same namespace
+	// to use for pulling temporal images from registries.
+	// +optional
+	ImagePullSecrets []corev1.LocalObjectReference `json:"imagePullSecrets,omitempty"`
+	// TemporalServer connection details.
+	TemporalConnection *TemporalConnectionSpec `json:"temporalConnection"`
+}
+
+// TemporalConnectionSpec defines the attributes for connecting to a Temporal server.
+type TemporalConnectionSpec struct {
+	// FQDN of the temporal frontend service endpoint.
+	URL string `json:"url"`
+	// Port where the temporal frontend service is listening.
+	Port *int `json:"port"`
+	// Namespace that worker will poll.
+	Namespace string `json:"namespace"`
+}
+
+// TemporalAppWorkerStatus defines the observed state of TemporalAppWorker
+type TemporalAppWorkerStatus struct {
+	// Conditions represent the latest available observations of the Cluster state.
+	Conditions []metav1.Condition `json:"conditions"`
+	// Number of desired replicas. Default to 1.
+	// +kubebuilder:validation:Minimum=1
+	// +optional
+	Replicas *int32 `json:"replicas"`
+}
+
+// +genclient
+// +genclient:Namespaced
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+// +kubebuilder:printcolumn:name="Ready",type="string",JSONPath=".status.conditions[?(@.type == 'Ready')].status"
+// +kubebuilder:printcolumn:name="ReconcileSuccess",type="string",JSONPath=".status.conditions[?(@.type == 'ReconcileSuccess')].status"
+// +kubebuilder:printcolumn:name="Age",type="date",JSONPath=".metadata.creationTimestamp"
+
+// TemporalAppWorker is the Schema for the temporalappworkers API
+type TemporalAppWorker struct {
```

Comment:
I like the idea of not calling this CRD `TemporalWorker`, as said in the issue we have to make the distinction between users workers and the temporal worker.
I took some time on the temporal documentation, and what do you think of `TemporalWorkerProcess` ? https://docs.temporal.io/workers#worker-process

Reply:
@alexandrevilain I would be ok with TemporalWorkerProcess but I still like TemporalAppWorker better. This is actually what we refer to workers when talking to Temporal customers "Application Workers". I like the word application or 'app' because it implies that is it an application which is exactly what a worker produces. I think WorkerProcess is a bit vague and also anything can be a process but an application implies this is something customer builds or creates which I like. I will leave it up to you for final decision, I just wanted to make my case on why AppWorker. Please let me know?

Reply:
@alexandrevilain yeah at first I was still gonna make a case for TemporalAppWorker, I like it better and we do in talks with customers talk about "application workers". However as you pointed out our docs refer to TemporalWorkerProcess so I think being consistent is important. I will change the name. We can always change it in future but for now agree lets be consistent with docs!